File: src/main/java/trivia/QuestionDeck.java
package trivia;

import java.util.LinkedList;
import java.util.Map;
import java.util.Queue;
import java.util.LinkedHashMap;

public class QuestionDeck {
    private final Map<String, Queue<String>> questionsByCategory = new LinkedHashMap<>();

    public QuestionDeck(int initialQuestions) {
        questionsByCategory.put("Pop", new LinkedList<>());
        questionsByCategory.put("Science", new LinkedList<>());
        questionsByCategory.put("Sports", new LinkedList<>());
        questionsByCategory.put("Rock", new LinkedList<>());

        for (int i = 0; i < initialQuestions; i++) {
            questionsByCategory.get("Pop").add("Pop Question " + i);
            questionsByCategory.get("Science").add("Science Question " + i);
            questionsByCategory.get("Sports").add("Sports Question " + i);
            questionsByCategory.get("Rock").add("Rock Question " + i);
        }
    }

    public String nextQuestion(String category) {
        return questionsByCategory.get(category).poll();
    }
}

File: python/question_deck.py
from collections import deque

class QuestionDeck:
    def __init__(self, initial_questions: int):
        self.questions_by_category = {
            "Pop": deque(),
            "Science": deque(),
            "Sports": deque(),
            "Rock": deque()
        }

        for i in range(initial_questions):
            self.questions_by_category["Pop"].append(f"Pop Question {i}")
            self.questions_by_category["Science"].append(f"Science Question {i}")
            self.questions_by_category["Sports"].append(f"Sports Question {i}")
            self.questions_by_category["Rock"].append(f"Rock Question {i}")

    def next_question(self, category: str) -> str:
        return self.questions_by_category[category].popleft()
